tests/unit/test_models.py

```diff
@@ -358,11 +358,13 @@ def test_non_converged_solution(self):
                 "Voltage [V]": np.zeros(100),
             }
         )
-
         problem = pybop.FittingProblem(model, parameters=parameters, dataset=dataset)
-        res = problem.evaluate([-0.2, -0.2])
-        _, res_grad = problem.evaluateS1([-0.2, -0.2])
+
+        # Simulate the DFN with active material values of 0.
+        # The solutions will not change as the solver will not converge.
```

Comment:
Please clarify if you are testing that both solutions fail, rather than old solutions being returned when the solver does not converge.